### PR TITLE
add(Text2Image): Add dev docs for TextToImage OCP and OCS API + admin docs for TextToImage

### PR DESCRIPTION
**Pull request by marcelklehr**
### ☑️ Resolves

* Docs for https://github.com/nextcloud/server/pull/40326

### 🖼️ Screenshots

| OCS API| OCP API | Admin manual|
|---|---|---|
| image | image | image |

### CODE REVIEW

File: developer_manual/digging_deeper/text2image.rst [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+Consuming the Text-To-Image API
+-------------------------------
+
+To consume the Text-To-Image API, you will need to :ref:`inject<dependency-injection>` ``\OCP\TextToImage\IManager``. This manager offers the following methods:
+
+ * ``hasProviders()`` This method returns a boolean which indicates if any providers have been registered. If this is false you cannot use the image generation feature.
+ * ``runTask(Task $task)`` This method provides the actual functionality. The task is defined using the Task class. This method runs the task synchronously, so depending on the implementation it is uncertain how long it will take (between 3s and several hours).
+ * ``scheduleTask(Task $task)`` This method also runs a task, but asynchronously in a background job. The task is defined using the Task class.
+ * ``runOrScheduleTask(Task $task)`` This method also runs a task, but fist checks the expected runtime of the provider to be used. If the runtime fits inside the available processing time for the current request the task is run synchronously, otherwise it is scheduled as a background job. The task is defined using the Task class.
```
Comment (MB-Finski) [inline developer_manual/digging_deeper/text2image.rst:19]: How to handle the outcome of runOrScheduleTask is somewhat unclear based on the docs/dockblocks as it requires different logic based on whether the task was run or scheduled. Would it make sense to provide a more info and/or a sample code snippet of how to deal with outputs of runTask/scheduleTask/runOrScheduleTask?

(Minor typo on this line: "fist")

File: developer_manual/client_apis/OCS/ocs-text2image-api.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+* Response:
+    - Status code:
+        + ``200 OK``
+        + ``412 Precondition Failed`` - When the task type is not available currently
+        + ``429 Too Many Requests`` - When the rate limiting was exceeded
+
+    - Data:
+        + ``id`` - Only provided in case of ``200 OK``, the assigned task id, int
+        + ``input`` - Only provided in case of ``200 OK``, the task input, string
+        + ``status`` - Only provided in case of ``200 OK``, the current task status, int, see backend API
```
Comment (MB-Finski) [inline developer_manual/client_apis/OCS/ocs-text2image-api.rst:68]: I think you've added a lot of useful cross refs. Adding cross refs to the correct subsections for the occurences of "see backend API" in this file could save some time when figuring out how to handle the endpoint outputs...?

File: developer_manual/digging_deeper/text2image.rst (hunk 1 above)
Comment (marcelklehr) [reply developer_manual/digging_deeper/text2image.rst:19]: Yeah, good idea!

File: developer_manual/client_apis/OCS/ocs-text2image-api.rst (hunk 2 above)
Comment (marcelklehr) [reply developer_manual/client_apis/OCS/ocs-text2image-api.rst:68]: Good idea!

**Review by MB-Finski — APPROVED** [review]
I think it's really clear and concise. Should make fast work of both consuming the api or implementing a provider.

**Pull request merged**